ui/selectors/multichain.ts

```diff
@@ -97,18 +109,21 @@ export function getMultichainNetwork(
 // currency will be BTC..
 
 export function getMultichainIsEvm(state: MultichainState) {
-  const selectedAccount = getSelectedInternalAccount(state);
-
-  // There are no selected account during onboarding. we default to the current EVM provider.
-  return !selectedAccount || isEvmAccountType(selectedAccount.type);
+  const isOnboarded = getCompletedOnboarding(state);
+  // Selected account is not available during onboarding (this is used in
+  // the AppHeader)
+  const selectedAccount = getMaybeSelectedInternalAccount(state);
+
+  // There are no selected account during onboarding. we default to the original EVM behavior.
+  return (
+    !isOnboarded || !selectedAccount || isEvmAccountType(selectedAccount.type)
+  );
 }
 
-export function getMultichainProviderConfig(
-  state: MultichainState,
-): ProviderConfig | MultichainProviderConfig {
-  return getMultichainIsEvm(state)
-    ? getProviderConfig(state)
-    : getMultichainNetwork(state).network;
+export function getMultichainProviderConfig(state: MultichainState) {
+  // Naming is a bit confusing here, but we do use the `network` field that is actually
+  // a `{Multichain,}ProviderConfig`
+  return getMultichainNetwork(state).network;
 }
```

Comment:
We should probably use JSDocs to provide these explanations. For example,

```suggestion
/**
 * Retrieves the provider configuration for a multichain network.
 * 
 * This function extracts the `network` field from the result of `getMultichainNetwork(state)`,
 * which is expected to be a `MultichainProviderConfig` object. The naming might suggest that
 * it returns a network, but it actually returns a provider configuration specific to a multichain setup.
 *
 */
export function getMultichainProviderConfig(state: MultichainState) {
  return getMultichainNetwork(state).network;
}
```